src/coreComponents/mesh/generators/CellBlockManager.cpp

```diff
@@ -776,4 +802,398 @@ std::map< string, SortedArray< localIndex > > & CellBlockManager::getNodeSets()
+    {
+      for( int q2=0; q2<numNodesPerEdge; q2++ )
+      {
+        localIndex nodeID;
+        std::array< localIndex, 6 > nodeKey = createNodeKey( v1, v2, v3, v4, q1, q2, order );
+        if( nodeIDs.count( nodeKey ) == 0 )
+        {
+          // this is an internal face node: create it
+          nodeID = localNodeID;
+          nodeIDs[ nodeKey ] = nodeID;
+          std::array< globalIndex, 6 > referenceOrientation = createNodeKey( gv1, gv2, gv3, gv4, q1, q2, order );
+          int gq1 = referenceOrientation[4];
+          int gq2 = referenceOrientation[5];
+          nodeLocalToGlobalNew[ nodeID ] = offset + faceLocalToGlobal[ iter_face ] * numInternalNodesPerFace + gq1* numInternalNodesPerEdge + gq2;
+          localNodeID++;
+        }
+        else
+        {
+          nodeID = nodeIDs[ nodeKey ];
+        }
+        faceToNodeMapNew[ iter_face ][ q2 + q1*numNodesPerEdge ] = nodeID;
+      }
+    }
+  }
+
+  // add all nodes to the target set "all"
+  highOrderMeshLevel.getNodeManager().resize( numLocalNodes );
+
+  SortedArray< localIndex > & allNodesSet = this->getNodeSets()[ "all" ];
+  allNodesSet.reserve( numLocalNodes );
+
+  for( localIndex iter_nodes=0; iter_nodes< numLocalNodes; ++iter_nodes )
+  {
+    allNodesSet.insert( iter_nodes );
+  }
+
+  /////////////////////////
+  // Elements
+  //////////////////////////
+
+  // also assign node coordinates using trilinear interpolation in th elements
+  arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const refPosSource = source.getNodeManager().referencePosition();
+  arrayView2d< real64, nodes::REFERENCE_POSITION_USD > refPosNew = this->getNodePositions();
+  refPosNew.setValues< parallelHostPolicy >( -1.0 );
+
+  real64 Xmesh[ numVerticesPerCell ][ 3 ] = { { } };
+  real64 X[ 3 ] = { { } };
+  array1d< real64 > glCoords = gaussLobattoPoints( order );
+  localIndex elemMeshVertices[ numVerticesPerCell ] = { };
+  offset = maxVertexGlobalID + maxEdgeGlobalID * numInternalNodesPerEdge + maxFaceGlobalID * numInternalNodesPerFace;
+  std::array< localIndex, 6 > const nullKey = std::array< localIndex, 6 >{ -1, -1, -1, -1, -1, -1 };
+  source.getElemManager().forElementRegions< CellElementRegion >( [&]( CellElementRegion const & sourceRegion )
+  {
+    // create element region with the same name as source element region "Region"
+    CellElementRegion & region = *(dynamic_cast< CellElementRegion * >( highOrderMeshLevel.getElemManager().createChild( sourceRegion.getCatalogName(),
+                                                                                                                         sourceRegion.getName() ) ) );
+    // add cell block to the new element region with the same name as cell block name from source element region
+    region.addCellBlockNames( sourceRegion.getCellBlockNames() );
+
+    sourceRegion.forElementSubRegions< CellElementSubRegion >( [&]( CellElementSubRegion const & sourceSubRegion )
```

Comment:
This is not a geometrical concept, so it must be moved.
_Maybe_, it can even be removed because it should not change a lot w.r.t. the current `CellBlockManagerABC` -> `*Managers` that is already in place.

Reply:
Thanks Thomas for the suggestion. Yes, all the non-geometrical related components (face normals, centers, region , domain boundary, etc.) has been removed from CellBlockManager.cpp, and they are handled at the MeshLevel constructor for high-order mesh-level now.